spec/isec2_windows.go

```diff
@@ -0,0 +1,108 @@
+package spec
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Songmu/retry"
+	"github.com/StackExchange/wmi"
+)
+
+// Win32ComputerSystemProduct is struct for WMI. SKUNumber is nil-able.
+type Win32ComputerSystemProduct struct {
+	Caption           string
+	Description       string
+	IdentifyingNumber string
+	Name              string
+	SKUNumber         *string
+	UUID              string
+	Vendor            string
+	Version           string
+}
+
+// If the OS is Windows, check UUID in WMI class Win32_ComputerSystemProduct first. If UUID seems to be EC2-ish, call the metadata API (up to 3 times).
+// ref. https://docs.aws.amazon.com/ja_jp/AWSEC2/latest/WindowsGuide/identify_ec2_instances.html
+func isEC2(ctx context.Context) bool {
+	var records []Win32ComputerSystemProduct
+	err := wmi.Query("SELECT * FROM Win32_ComputerSystemProduct", &records)
```

Comment:
Is collecting one column better than all columns in performance? If yes I think it's worth collecting only UUID column.

Reply:
Okay, will fix.